query/src/sql/sql_parser.rs

```diff
@@ -854,23 +854,108 @@ impl<'a> DfParser<'a> {
         Ok(DfStatement::DropStage(drop))
     }
 
+    fn parse_udf_parameters(&mut self) -> Result<Vec<String>, ParserError> {
+        let mut params = vec![];
+        let mut found_right_paren = false;
+        let mut expect_next_param = false;
+
+        self.parser.expect_token(&Token::LParen)?;
+        loop {
+            let next_token = self.parser.next_token();
+            if next_token == Token::EOF || next_token == Token::SemiColon {
+                break;
+            }
+            if next_token == Token::RParen {
+                found_right_paren = true;
+                break;
+            }
+
+            match next_token {
+                Token::Word(Word {
+                    value,
+                    quote_style,
+                    keyword,
+                }) => {
+                    if keyword != Keyword::NoKeyword {
+                        return parser_err!(format!(
+                            "Keyword can not be parameter, got: {}",
+                            value
+                        ));
+                    }
+                    if let Some(quote) = quote_style {
+                        return parser_err!(format!(
+                            "Quote is not allowed in parameters, remove: {}",
+                            quote
+                        ));
+                    }
+                    if params.contains(&value) {
+                        return parser_err!(format!(
+                            "Duplicate parameter is not allowed, keep only one: {}",
+                            &value
+                        ));
+                    }
+
+                    expect_next_param = false;
+                    params.push(value);
+                }
+                Token::Comma => {
+                    expect_next_param = true;
+                }
+                other => {
+                    return parser_err!(format!("Expect words or comma, but got: {}", other));
+                }
+            }
+        }
+
+        if !found_right_paren {
+            return parser_err!("Can not find complete parameters, `)` is missing");
+        }
+        if expect_next_param {
+            return parser_err!("Found a redundant `,` in the parameters");
+        }
+
+        Ok(params)
+    }
+
+    fn parse_udf_definition_expr(&mut self, until_token: Vec<&str>) -> Result<String, ParserError> {
+        // Match ->
+        self.parser.expect_token(&Token::Minus)?;
+        let next_token = self.parser.next_token_no_skip();
+        if next_token != Some(&Token::Gt) {
```

Comment:
self.parser.expect_token(&Token::Gt)?;

Reply:
Don't think so. `expect_token` will skip the whitespace, what we need here is `->`.